thrust/thrust/functional.h

```diff
@@ -1082,37 +893,7 @@ struct project2nd<void, void>
  *  \{
  */
 
-namespace detail
-{
-template <typename F>
-struct not_fun_t
-{
-  F f;
-
-  template <typename... Ts>
-  _CCCL_HOST_DEVICE auto
-  operator()(Ts&&... args) noexcept(noexcept(!f(std::forward<Ts>(args)...))) -> decltype(!f(std::forward<Ts>(args)...))
-  {
-    return !f(std::forward<Ts>(args)...);
-  }
-
-  template <typename... Ts>
-  _CCCL_HOST_DEVICE auto operator()(Ts&&... args) const
-    noexcept(noexcept(!f(std::forward<Ts>(args)...))) -> decltype(!f(std::forward<Ts>(args)...))
-  {
-    return !f(std::forward<Ts>(args)...);
-  }
-};
-} // namespace detail
-
-//! Takes a predicate (a callable returning bool) and returns a new predicate that returns the negated result.
-//! \see https://en.cppreference.com/w/cpp/utility/functional/not_fn
-// TODO(bgruber): alias to ::cuda::std::not_fn in C++17
-template <class F>
-_CCCL_HOST_DEVICE auto not_fn(F&& f) -> detail::not_fun_t<::cuda::std::decay_t<F>>
-{
-  return detail::not_fun_t<::cuda::std::decay_t<F>>{std::forward<F>(f)};
-}
+using ::cuda::std::not_fn;
```

Comment:
hm not_fn is a function template in cuda::std.

I am wondering why it is not being pulled in :thinking_face:

Reply:
Oh, we probably need to merge the C++17 upgrade first, since `cuda::std::not_fn` requires it.